src/framework/components/element/component.js

```diff
@@ -674,27 +679,33 @@ class ElementComponent extends Component {
      * The position of the pivot of the component relative to its anchor. Each value ranges from 0
      * to 1 where [0,0] is the bottom left and [1,1] is the top right.
      *
-     * @type {Vec2}
+     * @example
+     * pc.app.root.findByName("Inventory").element.pivot = [Math.random() * 0.1, Math.random() * 0.1];
+     * @example
+     * pc.app.root.findByName("Inventory").element.pivot = new pc.Vec2(Math.random() * 0.1, Math.random() * 0.1);
+     *
+     * @type {Vec2 | number[]}
      */
     set pivot(value) {
-        const prevX = this._pivot.x;
-        const prevY = this._pivot.y;
+        const { pivot, margin } = this;
+        const prevX = pivot.x;
+        const prevY = pivot.y;
 
         if (value instanceof Vec2) {
-            this._pivot.set(value.x, value.y);
+            pivot.copy(value);
         } else {
-            this._pivot.set(value[0], value[1]);
+            pivot.set(...value);
```

Comment:
and the same here.

Reply:
Same here, the ES5 version is just something like:
```js
value = [1, 2, 3];
console.log.apply(null, value);
```
Is it okay to keep? I would argue it reduces the amount of stack pollution / needless array unpacking

Reply:
I still believe the original code would run faster, but this is not critical path so it's ok.